src/test/java/com/n1analytics/paillier/PaillierEncryptedNumberTest.java

```diff
@@ -1151,6 +1152,46 @@ public void testGetThisThingInSafe() {
         assertTrue(safeEN.getSafeEncryptedNumber().isSafe);
         assertEquals(safeEN.ciphertext, safeEN.getSafeEncryptedNumber().ciphertext);
     }
+
+    @Test
+    public void testJavaSerialization () throws IOException, ClassNotFoundException {
+      EncryptedNumber unSafeEN = context.encrypt(context.encode(1.01, 1e-8));
+      assertFalse(unSafeEN.isSafe);
+      EncryptedNumber safeEN = unSafeEN.getSafeEncryptedNumber();
+      assertTrue(safeEN.isSafe);
+      assertNotEquals(unSafeEN.ciphertext, safeEN.ciphertext);
+
+      //test writing and reading safe en
+      String file = "./temp";
```

Comment:
You should use `File.createTempFile()` to create temporary files, and call `myfile.deleteOnExit()` to make sure it's removed.

Reply:
Fixed in latest commit. Feel free to let me know if there is further problems or questions:)